src/cmd/build.js

```diff
@@ -45,11 +45,20 @@ export type LocalizedNameParams = {
   manifestData: ExtensionManifest,
 }
 
+// This defines the _locales/messages.json type. See:
+// https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Internationalization#Providing_localized_strings_in__locales
+type LocalizedMessageData = {
+  [messageName: string]: {
+    description: string,
+    message: string,
+  },
+}
+
 export async function getDefaultLocalizedName(
   {messageFile, manifestData}: LocalizedNameParams
 ): Promise<string> {
 
-  let messageData: string | Buffer;
+  let messageData: LocalizedMessageData;
```

Comment:
@kumar303 :+1:

I though that it was interesting to use this as an opportunity to figure out why flow didn't caught this before, and why it is complaining now.

I played a bit using `flow type-at-pos filename row col` and this is what was probably happening:

- on `let messageData: string | Buffer;`, message is a union type

- on `messageData = JSON.parse(...)`, messageData is of type `any`, because it is the type of the JSON.parse returned value

- with `if (!(messageData[messageName] && messageData[messageName].message)) { ... } else { ... }` we are forcing flow to "believe us when we say that in the `else` branch messageData has a property named `messageName` and it is an object which contains a `message` property"

But we were not saying anything about the type of the `message` property, and while the previous version of flow wasn't checking that the type of the replace `callback` returned value was actually a string type, the new version is inferencing the type correctly and detected that `message` is an unknown type while flow expects a string value to be returned (it is completely possible that the declarations of the `String.prototype.replace` wasn't declared with this level of details, or that flow type inference is better in the new version).